core/src/syn/parser/stmt/define.rs

```diff
@@ -186,6 +187,33 @@ impl Parser<'_> {
 					self.pop_peek();
 					res.permissions = ctx.run(|ctx| self.parse_permission_value(ctx)).await?;
 				}
+				t!("AS") => {
+					self.pop_peek();
+					let base = if self.eat(t!("ROOT")) {
+						Base::Root
+					} else if self.eat(t!("NAMESPACE")) {
+						Base::Ns
+					} else if self.eat(t!("DATABASE")) {
+						Base::Db
+					} else {
+						Base::Db // DATABASE is default level.
+					};
+					let kind = if self.eat(t!("ROLES")) {
```

Comment:
After the fix implemented in #5092 is merged, we would also need to [check against the existent roles](https://github.com/surrealdb/surrealdb/pull/5092/files#diff-347660d93a7a78de225183474aff64ca82597eb5cb57a54e2c31e523620b9c11R239-R259) here.